### PR TITLE
GNU GRUB 2 for i386-elf

### PR DESCRIPTION
**Pull request by MRNIU**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by pcr910303** [comment]
bump :-)

**Comment by MRNIU** [comment]
@pcr910303 
It's a coincidence. Are you work on cross compile? Maybe you need them:
i386-elf-gcc, i386-elf-binutils, i386-elf-gdb, of course i386-elf-grub.

**Comment by Varriount** [comment]
@MRNIU i386-elf-grub doesn't appear to be a currently existing package.

**Comment by MRNIU** [comment]
@Varriount Yes, but you can use it by `brew install i386-elf-grub.rb`.

**Comment by MRNIU** [comment]
@fxcoudert Hey, could you please merge this pr? It is been so long.

File: Formula/i386-elf-grub.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                             "TARGET_STRIP=i386-elf-strip",
+                             "TARGET_NM=i386-elf-nm",
+                             "TARGET_RANLIB=i386-elf-ranlib"
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    assert_match "i386-elf-grub-glue-efi (GRUB) 2.02", shell_output("#{Formula["i386-elf-grub"].bin}/i386-elf-grub-glue-efi --version")
```
Comment (fxcoudert) [inline Formula/i386-elf-grub.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/i386-elf-grub.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class I386ElfGrub < Formula
+  desc "GNU GRUB 2 for i386-elf"
+  homepage "https://www.gnu.org/software/grub/"
+  url "https://ftp.gnu.org/gnu/grub/grub-2.02.tar.xz"
+  sha256 "810b3798d316394f94096ec2797909dbf23c858e48f7b3830826b8daa06b7b0f"
+
+  depends_on "i386-elf-gcc"
```
Comment (fxcoudert) [inline Formula/i386-elf-grub.rb]: Is this a build-time or run-time dependency?

**Review by fxcoudert — CHANGES_REQUESTED** [review]
- What's the use case of GRUB on macOS?
- Is this 32-bit specific, or can it handle 64-bit (aka x86_64)?

**Comment by MRNIU** [comment]
@fxcoudert 
1. It allows users to perform operations that require grub support on the mac, such as making image files that are booted using grub, verifying existing files, etc.
2. x86_64
3. The function of grub needs to have an additional image file to verify, and we can't seem to do it.
4. build-time

**Comment by fxcoudert** [comment]
- Mark the build-time dependency with `=> :build`
- Use `resource` to download a small image for testing

File: Formula/i386-elf-grub.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                             "TARGET_STRIP=i386-elf-strip",
+                             "TARGET_NM=i386-elf-nm",
+                             "TARGET_RANLIB=i386-elf-ranlib"
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    resource ("image") do
```
Comment (fxcoudert) [inline Formula/i386-elf-grub.rb]: Please move the `resource do` block above the `install` method.

File: Formula/i386-elf-grub.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        url "https://raw.githubusercontent.com/MRNIU/tools/master/kernel.kernel"
+        sha256 "ac0e87c8a612fdfd4ce4c7f0a2161b5d156972d757c781cc5690735303861abf"
+    end
+    resource("image").stage { assert_match "", shell_output("#{Formula["i386-elf-grub"].bin}/i386-elf-grub-file --is-x86-multiboot2 kernel.kernel") }
```
Comment (fxcoudert) [inline Formula/i386-elf-grub.rb]: `assert_match ""` will always succeed, won't it? It's not a very good test.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/i386-elf-grub.rb (hunk 4 above)
Comment (MRNIU) [reply Formula/i386-elf-grub.rb]: Sorry, my apologize.

File: Formula/i386-elf-grub.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                             "TARGET_STRIP=i386-elf-strip",
+                             "TARGET_NM=i386-elf-nm",
+                             "TARGET_RANLIB=i386-elf-ranlib"
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    if resource("image").stage { shell_output("#{Formula["i386-elf-grub"].bin}/i386-elf-grub-file --is-x86-multiboot2 kernel.kernel") }
```
Comment (fxcoudert) [inline Formula/i386-elf-grub.rb]:
```suggestion
    resource("image").stage do
      system bin/"i386-elf-grub-file", "--is-x86-multiboot2", "kernel.kernel"
    done
```

**Comment by fxcoudert** [comment]
You have force-pushed over my changes. Please adapt the test as suggested, and squash all commits into a single one, named `i386-elf-grub 2.02 (new formula)`

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @MRNIU for the pull request!

**Pull request closed without merging**